kuma/wiki/templatetags/ssr.py

```diff
@@ -1,15 +1,32 @@
 from __future__ import print_function
 
 import json
+import os
 
 import requests
 import requests.exceptions
 from django.conf import settings
+from django.utils import lru_cache
 from django_jinja import library
 
+@lru_cache.lru_cache()
+def get_localization_data(locale):
+    """
+    Read the frontend string catalog for the specified locale, parse
+    it as JSON, and return the resulting dict. The returned values
+    are cached so that we don't have to read files all the time.
+    """
+    try:
+        path = os.path.join(settings.BASE_DIR,
+                            'static', 'jsi18n', locale, 'react.json')
+        with open(path, 'r') as f:
+            return json.load(f)
+    except Exception as e:
```

Comment:
I'd rather not do this. In fact, I don't think we should accept any exceptions. Why wouldn't the file be there and valid JSON? 

If someone visits (beta.)developer.m.o/en-OS/docs/Foo isn't there already some code that checks `en-OS` against `settings.ACCEPTED_LOCALES` and throws out the request with a 404 Not Found, well before we get this deep. 

If the .json file doesn't exist or is corrupted, I'd rather we find out about it sooner. 

I could accept a broad and brutal defensive approach if this was a lot more fragile and depended on things like network or some other IO. And if our web server ephemeral disk is so fragile that reading disk IO is that fragile I still don't want to use this technique because that `return None` will get stuck in the Python processor memory until the Gunicorn worker is recycled (which is rare). 

Also, considering that this is for the beta site, I'd sooner rather than later learn about potential unexpected errors here.

Reply:
To be clear, I'm saying this in the argument strength that the context is the beta site. There, I think we can afford more learning-by-errors.